src/lib/parser/core/PokemonSaveParser.ts

```diff
@@ -624,6 +628,150 @@ export class PokemonSaveParser {
+  async watch (options: {
+    onPartyChange?: (partyPokemon: PokemonBase[]) => void,
+    onError?: (error: Error) => void
+  } = {}): Promise<void> {
+    if (!this.isMemoryMode || !this.webSocketClient) {
+      throw new Error('Watch mode only available in memory mode (WebSocket connection)')
+    }
+
+    if (this.isWatching) {
+      throw new Error('Already watching memory. Call stopWatching() first.')
+    }
+
+    // Add callback to listeners if provided
+    if (options.onPartyChange) {
+      this.watchListeners.push(options.onPartyChange)
+    }
+
+    // Set up memory watching for party data regions
+    const partyRegions = [
+      { address: 0x20244e9, size: 7 }, // Party count + context  
+      { address: 0x20244ec, size: 600 }, // Full party data (6 * 100 bytes)
+    ]
+
+    try {
+      // Configure and start watching
+      this.webSocketClient.configureSharedBuffer({
+        preloadRegions: partyRegions
+      })
+      
+      await this.webSocketClient.startWatchingPreloadRegions()
+
+      // Set up memory change listener
+      const memoryChangeListener = async (address: number, _size: number, _data: Uint8Array) => {
+        try {
+          // Only process changes to party-related memory regions
+          if (address === 0x20244e9 || address === 0x20244ec) {
+            await this.handlePartyDataChange()
+          }
+        } catch (error) {
+          if (options.onError) {
+            options.onError(error instanceof Error ? error : new Error(String(error)))
+          }
+        }
+      }
+
+      this.webSocketClient.addMemoryChangeListener(memoryChangeListener)
+      this.isWatching = true
+
+      // Load initial data to establish baseline
+      await this.handlePartyDataChange()
+
+    } catch (error) {
+      if (options.onError) {
+        options.onError(error instanceof Error ? error : new Error(String(error)))
+      } else {
+        throw error
+      }
+    }
+  }
+
+  /**
+   * Handle party data changes and notify listeners
+   */
+  private async handlePartyDataChange (): Promise<void> {
+    if (!this.webSocketClient) return
+
+    try {
+      const partyPokemon = await this.parsePartyPokemon()
+      
+      // Create a hash of the party data to detect actual changes
+      const dataHash = JSON.stringify(
+        partyPokemon.map(p => ({
+          species: p.speciesId,
+          level: p.level,
+          hp: p.currentHp,
+          nickname: p.nickname,
+        }))
+      )
+
+      // Only notify listeners if party data actually changed
+      if (dataHash !== this.lastDataHash) {
+        this.lastDataHash = dataHash
+        
+        // Notify all listeners
+        for (const listener of this.watchListeners) {
+          try {
+            listener(partyPokemon)
+          } catch (error) {
+            console.error('Watch listener error:', error)
+          }
+        }
+      }
+    } catch (error) {
+      console.error('Error handling party data change:', error)
+    }
```

Comment:
we should not have to check for this here. the lua http-server should only send data when it changed